hooks/pre-commit

```diff
@@ -26,6 +26,34 @@ else
    echo "$(tput setaf 1)* No$(tput sgr 0)"
 fi
 
+npm run pretty-check >/dev/null 2>/dev/null
```

Comment:
Other SDKs also have a way of checking for circular dependencies, we should add that here

Reply:
its added 
![image](https://user-images.githubusercontent.com/13642906/191427484-ab905008-a59f-41fe-bed9-919569b35e86.png)